### PR TITLE
chore(anomaly-detection) Use sensitivity flag

### PR DESCRIPTION
**Pull request by ram-senth**
This PR applies the sensitivity flag to the the anomaly detection logic.

### CODE REVIEW

File: src/seer/anomaly_detection/detectors/mp_scorers.py [hunk 1]
```diff
@@ -199,37 +235,34 @@ def stream_score(
...
+        IQR = Q3 - Q1
+        threshold = Q3 + (1.5 * IQR)
+        return threshold
+
+    def _to_flag(self, mp_dist: np.float64, threshold: float):
+        if np.isnan(mp_dist) or mp_dist <= threshold:
             return "none"
-        if mp_dist < threshold_upper:
-            return "anomaly_lower_confidence"
         return "anomaly_higher_confidence"
```
Comment (aayush-se) [inline src/seer/anomaly_detection/detectors/mp_scorers.py:266]: If we are no longer doing lower vs higher confidence, can we just rename this to "anomaly"

File: src/seer/anomaly_detection/detectors/mp_scorers.py [hunk 2]
```diff
@@ -53,14 +54,43 @@ class LowVarianceScorer(MPScorer):
...
+        description="Lower and upper bounds for high sensitivity",
     )
 
     def _to_flag_and_score(
-        self, val: np.float64, ts_mean: np.float64
+        self,
+        val: np.float64,
+        ts_mean: np.float64,
+        sensitivity: Sensitivities,
+        direction: Directions,
```
Comment (aayush-se) [inline src/seer/anomaly_detection/detectors/mp_scorers.py:81]: Is direction something that will be implemented later?

**Review by aayush-se — APPROVED** [review]
Looks great! Just need to fix the Sentry tag and should be good to merge

File: src/seer/anomaly_detection/detectors/mp_scorers.py [hunk 3]
```diff
@@ -99,16 +129,34 @@ def stream_score(
...
 class MPIQRScorer(MPScorer):
+    """
+    This class scores anomalies using the interquartile range of the matrix profile distances.
+    """
+
+    percentiles: Dict[Sensitivities, Tuple[float, float]] = Field(
+        {
+            # High sensitivity = more anomalies + higher false positives
+            # Data point outside of bottom 65% of the MP distances considered anomalous
+            "high": [0.25, 0.75],
```
Comment (trillville) [inline src/seer/anomaly_detection/detectors/mp_scorers.py:150]: does "medium" correspond to "default" in terms of what looks to be ideal behavior? is there some validation results we can look at comparing them with this latest logic?

File: src/seer/anomaly_detection/detectors/normalizers.py [hunk 4]
```diff
@@ -24,3 +24,44 @@ def normalize(self, array: npt.NDArray) -> npt.NDArray:
             else:
                 return np.full_like(array, 1.0)
         return (array - np.min(array)) / (np.max(array) - np.min(array))
+
+
+# class MinMaxStreamNormalizer:
```
Comment (trillville) [inline src/seer/anomaly_detection/detectors/normalizers.py:29]: why is this added but commented out?

**Review by trillville — APPROVED** [review]

File: src/seer/anomaly_detection/detectors/mp_scorers.py (hunk 1 above)
Comment (ram-senth) [reply src/seer/anomaly_detection/detectors/mp_scorers.py:266]: That will need changes from alerting system as well as it is part of the API. Also, I still am hopeful that we will be able to revive this part.

File: src/seer/anomaly_detection/detectors/mp_scorers.py (hunk 2 above)
Comment (ram-senth) [reply src/seer/anomaly_detection/detectors/mp_scorers.py:81]: I already have the changes done for supporting direction. Once this PR lands will have that out.

File: src/seer/anomaly_detection/detectors/mp_scorers.py (hunk 3 above)
Comment (ram-senth) [reply src/seer/anomaly_detection/detectors/mp_scorers.py:150]: Yes the validation results are in the brownbag notebook. Here is one example of how the flag makes a difference:
High Sensitivity:
image

Medium Sensitivity:
image

Low Sensitivity:
image

File: src/seer/anomaly_detection/detectors/normalizers.py (hunk 4 above)
Comment (ram-senth) [reply src/seer/anomaly_detection/detectors/normalizers.py:29]: I was experimenting to see if normalizing the time series and/or the matrix profile improves accuracy. It did not help much with the naive implementation. I still want to keep this around for more detailed experimentation in the future.

File: src/seer/anomaly_detection/detectors/mp_scorers.py [hunk 5]
```diff
@@ -72,18 +102,18 @@ def batch_score(
...
         window_size: int,
     ) -> Optional[FlagsAndScores]:
-        ts_variance = ts.var()
         ts_mean = ts.mean()
         scores = []
         flags = []
-        if ts_variance > self.variance_threshold:
+        if ts.std() > self.std_threshold:
             sentry_sdk.set_tag(AnomalyDetectionTags.LOW_VARIANCE_TS, 0)
             return None
```
[the remark this answers is not in the document]
Comment (ram-senth) [reply src/seer/anomaly_detection/detectors/mp_scorers.py:110]: Changing it to `LOW_VARIATION_TS`.
